src/or/rendcache.c

```diff
@@ -696,9 +720,59 @@ rend_cache_store_v2_desc_as_dir(const char *desc)
       rep_hist_stored_maybe_new_hs(e->parsed->pk);
     }
 
+    if (options->EnablePrivCount) {
+      /* A list of fields in the v2 descriptor is available at:
+       * https://gitweb.torproject.org/torspec.git/tree/rend-spec.txt#n218
+       *
+       * The v2 onion address or permanent-id is in the descriptor but is not
+       * parsed.
+       * Use desc_id_base32 instead: there is one desc_id per service address,
+       * per replica (x2), per day for no auth and basic auth.
+       * For stealth auth, each client key has a different desc_id, per service
+       * address, per replica, per day.
+       */
+
+      /* Successful stores */
+      tor_assert(!failure_reason);
+      control_event_privcount_hsdir_cache_stored(HS_VERSION_TWO,
+                                               /* cache info */
+                                               has_existing_cache_entry,
+                                               1, /* added to cache */
+                                               (has_existing_cache_entry
+                                                ? "updated"
+                                                : "new"),
+                                               /* descriptor info */
```

Comment:
I agree it's ugly, it's a compromise between make check-spaces and my editor's automatic indenting. I'll try to do better next time.